File: backoffice/src/main/java/com/elevenof/backoffice/model/Event.java
package com.elevenof.backoffice.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Entity
@Table(name = "events")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Event {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(length = 500)
    private String picture;

    @Column(length = 500)
    private String shortContent;

    @Column(columnDefinition = "TEXT")
    private String content;

    @Column(nullable = false)
    private LocalDate startDate;

    @Column
    private LocalTime startTime;

    @Column(nullable = false)
    private LocalDate endDate;

    @Column
    private LocalTime endTime;

    @Column(length = 300)
    private String location;

    @ManyToOne
    @JoinColumn(name = "province_id")
    private Province province;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 20)
    private EventStatus status;

    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @Column(nullable = false)
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }

    public enum EventStatus {
        PLAN,              // Đang lên kế hoạch
        OPEN_REGISTER,     // Đang mở đăng ký
        CLOSE_REGISTER,    // Đóng đăng ký
        COMPLETE,          // Hoàn thành
        CANCELLED,         // Hủy bỏ
        DELETED            // Xóa
    }
}

File: frontend-app/src/types/event.ts
export type EventStatus =
  | "PLAN"
  | "OPEN_REGISTER"
  | "CLOSE_REGISTER"
  | "COMPLETE"
  | "CANCELLED"
  | "DELETED";

export interface EventListItem {
  id: number;
  title: string;
  picture: string | null;
  shortContent: string | null;
  startDate: string; // ISO date
  startTime: string | null; // HH:mm format
  endDate: string; // ISO date
  endTime: string | null; // HH:mm format
  location: string | null;
  provinceName: string | null;
  status: EventStatus;
  participantCount: number;
}

export interface EventDetail {
  id: number;
  title: string;
  picture: string | null;
  shortContent: string | null;
  content: string | null;
  startDate: string;
  startTime: string | null;
  endDate: string;
  endTime: string | null;
  location: string | null;
  provinceName: string | null;
  status: EventStatus;
  participantCount: number;
}

export interface EventsFilters {
  search: string;
  status: EventStatus | "";
  provinceId: string;
  sortBy: string;
  sortOrder: "asc" | "desc";
}

export function getStatusDisplayName(status: EventStatus): string {
  const statusMap: Record<EventStatus, string> = {
    PLAN: "Đang lên kế hoạch",
    OPEN_REGISTER: "Đang mở đăng ký",
    CLOSE_REGISTER: "Đóng đăng ký",
    COMPLETE: "Hoàn thành",
    CANCELLED: "Hủy bỏ",
    DELETED: "Đã xóa",
  };
  return statusMap[status] || status;
}

export function getStatusBadgeColor(status: EventStatus): string {
  const colorMap: Record<EventStatus, string> = {
    PLAN: "bg-gray-500",
    OPEN_REGISTER: "bg-green-600",
    CLOSE_REGISTER: "bg-yellow-600",
    COMPLETE: "bg-blue-500",
    CANCELLED: "bg-red-600",
    DELETED: "bg-gray-400",
  };
  return colorMap[status] || "bg-gray-500";
}
